src/ui/metadata_panel.py

```diff
@@ -555,14 +560,18 @@ def save_settings(self):
         # Write out changes to stl_dir, part_dir, author, license
         # default_part_name is always "untitled.dat"
 
-        settings = [self.stl_dir, "untitled.dat", self.part_dir,
-                    self.author_text, self.license_text]
-        file_path = Util.path_conversion("assets/settings/user_settings.txt")
+        file_path = Util.path_conversion("assets/settings/user_settings.json")
         try:
-            with open(file_path, "w") as file:
-                for setting in settings:
-                    if setting is not None:
-                        print(setting, file=file)
+            with open(file_path, "r+") as file:
```

Comment:
For me `r+` is opening file and appending new json to old json. So, `json.dump` and `json.load` are giving `JSONDecodeError("Extra data", s, end)` error. Can we just open the file in `w` mode and add a new json instead of replacing older?